### ISSUE TITLE
Spike: gather requirements for ephemeral environments

### ISSUE THREAD

**Issue by katherine-black** [cloudigrade/cloudigrade#772]
In GitLab by @infinitewarp on Dec 16, 2020, 18:15

## Summary

We need to understand all that's required to make cloudigrade use ephemeral environments in OSDv4, presumably leveraging [Clowder](https://github.com/RedHatInsights/clowder/) and maybe [Bonfire](https://github.com/RedHatInsights/bonfire). Presumably other teams have already done this, and we can find some excellent reference examples.

## Timebox

3 days.

## Desired output

- [x] Ideally a reference example configs of a similar platform service that uses RDS and CloudWatch for logs.
- [x] Links to specific sections of documentation around what we need to change in e2e-deploy, app-interface, saas-templates, etc.
- [x] Answers to questions about availability of other services/components/etc we need. Is 3scale done? Is sources-api done? Is sources-ui and the rest of the web interface for sources done? Is Turnpike done or do we have proper access (port-forward? pod shell/exec/terminal?) to the pods?
- [x] Conversation with the team about what all this entails!
- [x] Skeleton issue defined with list of things we need to implement to make ephemeral environments a reality.

## Assumptions and Questions

- Are all the components available and ready for us? Check on 3scale; this was missing at one point.
- There's really no point in us actually building the ephemeral environments (yet) if sources-api itself is not already *completely* ready for ephemeral environment use.
- We already have several issues in the backlog that we need to complete before we can go fully ephemeral. For example, we need the issues for automatically setting up all the AWS things and for making more configs name/convention-based. @infinitewarp to go find and link these.

**Comment by katherine-black** [comment]
In GitLab by @mpierce on Dec 17, 2020, 11:40

Here are some recent resources relevant to this conversation:

[Clowder Platform Status](https://docs.google.com/spreadsheets/d/101p7bfe49k-7TzHIpAu3Ln9ro715mCWwdNuIyui4XI4/edit#gid=0)

[Migrating to Clowder docs](https://internal.cloud.redhat.com/docs/devprod/migration/)

[Ephemeral Environments docs](https://internal.cloud.redhat.com/docs/devprod/ephemeral/)

[Working with Clowder](https://docs.google.com/document/d/15mQrFmSw-nmi8AQ3Swt5Y1m55oo4_FFnbeDr9tUDhjU/edit#heading=h.9xu06i7s46u3)

**Comment by katherine-black** [comment]
In GitLab by @werwty on Jan 6, 2021, 13:33

unassigned @werwty

**Comment by katherine-black** [comment]
In GitLab by @werwty on Jan 25, 2021, 09:47

marked the task **Conversation with the team about what all this entails!** as completed

**Comment by katherine-black** [comment]
In GitLab by @werwty on Jan 25, 2021, 09:47

marked the task **Skeleton issue defined with list of things we need to implement to make ephemeral environments a reality.** as completed

**Comment by katherine-black** [comment]
In GitLab by @werwty on Jan 25, 2021, 16:30

> Ideally a reference example configs of a similar platform service that uses RDS and CloudWatch for logs.

- Each ephemeral environment will have their own database pod, no RDS/Cloudwatch


> Answers to questions about availability of other services/components/etc we need. Is 3scale done? Is sources-api done? Is sources-ui and the rest of the web interface for sources done? Is Turnpike done or do we have proper access (port-forward? pod shell/exec/terminal?) to the pods?

- 3 scale is done, but requires some manual intervention: https://internal.cloud.redhat.com/docs/devprod/ephemeral/03-frontend/#an-ephemeral-deployment-of-the-3scale-gateway
- Turnpike is not done, but we do have the ability to oc rsh/ port-forward
- Sources will likely not be done until post summit.

**Comment by katherine-black** [comment]
In GitLab by @katherine-black on Jan 26, 2021, 09:29

marked the task **Ideally a reference example configs of a similar platform service that uses RDS and CloudWatch for logs.** as completed

**Comment by katherine-black** [comment]
In GitLab by @katherine-black on Jan 26, 2021, 09:29

marked the task **Links to specific sections of documentation around what we need to change in e2e-deploy, app-interface, saas-templates, etc.** as completed

**Comment by katherine-black** [comment]
In GitLab by @katherine-black on Jan 26, 2021, 09:29

marked the task **Answers to questions about availability of other services/components/etc we need. Is 3scale done? Is sources-api done? Is sources-ui and the rest of the web interface for sources done? Is Turnpike done or do we have proper access (port-forward? pod shell/exec/terminal?) to the pods?** as completed

**Comment by katherine-black** [comment]
In GitLab by @katherine-black on Jan 26, 2021, 09:29

Thank you for gathering this information, closing.
